shared/libraries/config_protocol/include/config_protocol/config_client_input_port_impl.h

```diff
@@ -136,6 +138,45 @@ inline ErrCode ConfigClientInputPortImpl::assignSignal(ISignal* signal)
     return Super::connect(signal);
 }
 
+inline ErrCode INTERFACE_FUNC ConfigClientInputPortImpl::acceptsSignal(ISignal* signal, Bool* accepts)
+{
+    OPENDAQ_PARAM_NOT_NULL(signal);
+    OPENDAQ_PARAM_NOT_NULL(accepts);
+
+    return daqTry(
+        [this, &signal, &accepts]
+        {
+            // if (!this->deserializationComplete) // TODO remove ?
+            //    return Super::acceptsSignal(signal, accepts);
+
+            assert(clientComm->getConnected());  // TODO???
+
+            if (!(clientComm->getProtocolVersion() >= 3))
+                return makeErrorInfo(OPENDAQ_ERR_SIGNAL_NOT_ACCEPTED,
+                                     "Operation not supported by the protocol version currently in use");
+
+            const auto signalPtr = SignalPtr::Borrow(signal);
+            if (!isSignalFromTheSameComponentTree(signalPtr))
+                return makeErrorInfo(OPENDAQ_ERR_SIGNAL_NOT_ACCEPTED, "Signal is not from the same component tree");
```

Comment:
Same for this. The signal not accepted error is currently only meant for the `connect` call.